### PR TITLE
docs: add migration guide and notes for upcoming react 18 upgrade

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds some documentation about the upcoming react 18 upgrade

**Why do we need this feature?**

- so plugin devs are made aware as soon as possible and can test their plugins prior to grafana 10 release

**Who is this feature for?**

- plugin developers!

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/whatsnew/whats-new-in-v9-5.md
Comment (ashharrison90) [inline docs/sources/whatsnew/whats-new-in-v9-5.md:204]: waiting for approval on these docs internally before adding a forum post. once it's added, i'll update the link here

File: docs/sources/whatsnew/whats-new-in-v9-5.md
Comment (imatwawana) [inline docs/sources/whatsnew/whats-new-in-v9-5.md:203]:
```suggestion
- Test your plugins against one of the latest [grafana-dev docker images](https://hub.docker.com/r/grafana/grafana-dev/tags?page=1) (for example, [this one](https://hub.docker.com/layers/grafana/grafana-dev/10.0.0-111404pre/images/sha256-ac78acf54b44bd2ce7e68b796b1df47030da7f35e53b02bc3eec3f4de05f780f?context=explore))
```

**Review by imatwawana — APPROVED** [review]
I've made some style and wording edits, but otherwise LGTM!

**Review by mckn — COMMENTED** [review]
Overall, I think it looks great! What do you think about adding this to the [`migration-guide`](https://github.com/grafana/grafana/tree/main/docs/sources/developers/plugins/migration-guide) and then reference it here as well?

**Comment by ashharrison90** [comment]
@mckn think that sounds like a good idea 👍 is there a branch somewhere already for plugin changes needed for v10? if not i can create a folder on this branch following the same structure

File: docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md
Comment (ashharrison90) [inline docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:26]: waiting for approval on these docs internally before adding a forum post. once it's added, i'll update the link here

**Comment by ashharrison90** [comment]
@grafana/plugins-platform-frontend are we happy with these docs? also is there a way of running the docs locally? for me, `make docs` just hangs 😢

File: docs/sources/developers/plugins/migration-guide/_index.md
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/migration-guide/_index.md:28]: I doubt you'll need this as this was for redirects when we split out the sections from mega page

File: docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:18]: Given this is a doc about the major changes in Grafana 10 we can probably simplify this, suggestion: "Grafana 10 includes our upgrade to React 18..."

**Review by sympatheticmoose — APPROVED** [review]
Tested locally and LGTM, some minor suggestions in the review.

I also changed the PR title a little

**Pull request merged**
